test/letters/actions/letters.unit.spec.js

```diff
@@ -1,86 +1,495 @@
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.firstCall.args[0];
+        expect(action.type).to.equal(GET_ADDRESS_SUCCESS);
+      }).then(done, done);
+  });
+
+  it('dispatches GET_ADDRESS_FAILURE when GET fails', (done) => {
+    global.fetch.returns(Promise.reject({}));
+    const thunk = getMailingAddress();
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.firstCall.args[0];
+        expect(action.type).to.equal(GET_ADDRESS_FAILURE);
+      }).then(done, done);
+  });
+
+  it('dispatches GET_ADDRESS_FAILURE when response mangled', (done) => {
+    global.fetch.returns(Promise.resolve({
+      headers: { get: () => 'application/json' },
+      ok: true,
+      json: () => Promise.resolve({})
+    }));
+    const thunk = getMailingAddress();
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.firstCall.args[0];
+        expect(action.type).to.equal(GET_ADDRESS_FAILURE);
+      }).then(done, done);
+  });
+
+  // Note: not really sure we need to test this as long as the next test passes
+  it('dispatches with clone of response object (not original)', (done) => {
+    global.fetch.returns(Promise.resolve({
+      headers: { get: () => 'application/json' },
+      ok: true,
+      json: () => Promise.resolve(addressResponse)
+    }));
+    const thunk = getMailingAddress();
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.firstCall.args[0];
+        expect(action.data).to.not.equal(addressResponse);
+      }).then(done, done);
+  });
+
+  it('modifies military addresses', (done) => {
+    const militaryAddress = {
+      ...addressResponse.data.attributes.address,
+      type: 'MILITARY',
+      militaryPostOfficeTypeCode: 'APO',
+      militaryStateCode: 'AE'
+    };
 
-    thunk(dispatchSpy);
+    const militaryResponse = { ...addressResponse };
+    militaryResponse.data.attributes.address = {
+      ...addressResponse.data.attributes.address,
+      ...militaryAddress
+    };
+
+    global.fetch.returns(Promise.resolve({
+      headers: { get: () => 'application/json' },
+      ok: true,
+      json: () => Promise.resolve(militaryResponse)
+    }));
+    const thunk = getMailingAddress();
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.firstCall.args[0];
+        const { address } = action.data.data.attributes; // Actual
+        const { militaryPostOfficeTypeCode, militaryStateCode } = militaryAddress; // Test
+        expect(address.city).to.equal(militaryPostOfficeTypeCode);
+        expect(address.stateCode).to.equal(militaryStateCode);
+        expect(address.militaryPostOfficeTypeCode).to.be.undefined;
+        expect(address.militaryStateCode).to.be.undefined;
```

Comment:
Excellent! In your PR comment, you might mention that this closes https://github.com/department-of-veterans-affairs/vets.gov-team/issues/4961

Reply:
Oh right, nice.